File: 88.cpp
#include <bits/stdc++.h>
using namespace std;
void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
    if(!nums1.size() && nums2.size()) nums1 = nums2;
    vector<int> resp; 
    int i=0;
    int j=0;

    while (i < nums1.size() && j < nums2.size()){
        if(nums1[i] <= nums2[j] && i < nums1.size()-n){
            resp.push_back(nums1[i]);
            i++;
        } else {
            resp.push_back(nums2[j]);
            j++;
        }
    }
    while(i < nums1.size()){
        if(i < nums1.size()-n) resp.push_back(nums1[i]);
        i++; 
    }

    while(j < nums2.size()){
        resp.push_back(nums2[j]);
        j++;
    }
    nums1 = resp;
}
int main(){
    vector<int> nums1 = {1,2,3,0,0};
    vector<int> nums2 = {2,5,6};
    return 0;
}

File: Java/88.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int finalPointer  = m + n -1;
        while (n > 0 && m > 0) {
            if (nums1[m-1] < nums2[n-1]) {
                nums1[finalPointer] = nums2[n-1];
                n--;   
                finalPointer--;
            } else {
                nums1[finalPointer] = nums1[m-1];
                m--;
                finalPointer--;
            }
        }
        while (n > 0) {
            nums1[finalPointer] = nums2[n-1];
            finalPointer--;
            n--;
        }
        while (m > 0) {
            nums1[finalPointer] = nums1[m-1];
            finalPointer--;
            m--;
        } 
        // for (int i = 0; i < nums1.length; i++){
        //     System.out.println(nums1[i]);
        // } 
    }
}